### PR TITLE
[53] added package functionality for mailgun

### PR DESCRIPTION
**Pull request by malikkamranabid**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+      R5.out.log('Email not sent (on DEV)');
+    }
+
+    return message;
+  }
+};
+
+// Private Methods
+
+function send_queued (mailer) {
```
[the remark this answers is not in the document]
Comment (RaviPatel) [reply index.js:65]: Can we improve this?

**Review by RaviPatel — CHANGES_REQUESTED** [review]
We do also have a mailer_spec.js in games repo that we can add as a test

File: index.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+if (!global.R5) {
+  global.R5 = {
+    out: console
+  }
+}
+
+let mailer = require('mailgun-js');
+let config = {
+  domain:  'funnode.com',
+  key: '131133',
```
Comment (RaviPatel) [inline index.js]: what is this key?? this code won't work at all without the correct domain and key values provided by the user

**Review by RaviPatel — CHANGES_REQUESTED** [review]

File: test.js [hunk 3]
```diff
@@ -6,9 +6,16 @@ require('dotenv').config();
 var chai = require('chai');
 var assert = chai.assert;
 
+const config = {
+  domain:  'funnode.com',
+  key: '131133',
+  live: process.env.NODE_ENV === 'production'
+};
+
+
```
Comment (RaviPatel) [inline test.js]: minor: extra line

File: index.js [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+  live: process.env.NODE_ENV === 'production'
+};
+
+// Constructors
+
+function Mailer () {
```
Comment (RaviPatel) [inline index.js]: this needs to be updated to accept the config as a param instead of it being predefined

**Review by RaviPatel — CHANGES_REQUESTED** [review]

File: index.js [hunk 5]
```diff
@@ -10,18 +10,15 @@ if (!global.R5) {
 }
 
 let mailer = require('mailgun-js');
-let config = {
-  domain:  'funnode.com',
-  key: '131133',
-  live: process.env.NODE_ENV === 'production'
-};
+
```
Comment (RaviPatel) [inline index.js:13]: empty line

File: test.js [hunk 6]
```diff
@@ -16,7 +16,7 @@ const config = {
 
 describe('Mailer', function () {
   var mailer = new (require('./index.js'))(config);
-  var msg = { text: `test message` };
+  var msg = { text: `test message`};
```
Comment (RaviPatel) [inline test.js]: please add the space back as Standard.js will complain also

File: index.js [hunk 7]
```diff
@@ -10,18 +10,15 @@ if (!global.R5) {
...
 
 // Constructors
 
-function Mailer () {
+function Mailer (config) {
```
Comment (RaviPatel) [inline index.js]: since all the variables are mandatory, let's instead do this:
```
function Mailer (domain, key, live = false) {
```

**Review by RaviPatel — CHANGES_REQUESTED** [review]
almost done

File: index.js [hunk 8]
```diff
@@ -14,11 +14,13 @@ let mailer = require('mailgun-js');
 
 // Constructors
 
-function Mailer (config) {
-  this.config = config;
+function Mailer (domain, key, live = false) {
+  this.domain = domain;
```
Comment (RaviPatel) [inline index.js]: do these need to be saved in `this` if they're not being used again?

Comment (malikkamranabid) [reply index.js]: i cant access live variable without this as we are using it in a send function

Comment (RaviPatel) [reply index.js]: ok, but `domain` and `key` don't have to be saved, right?

Comment (malikkamranabid) [reply index.js]: yes right

**Review by RaviPatel — APPROVED** [review]

**Review by RaviPatel — APPROVED** [review]

**Pull request merged**